rpc/src/lib.rs

```diff
@@ -236,7 +236,8 @@ mod tests {
         let response: Result<u64> = client.request("some_subsystem_add", rpc_params!(2, 5)).await;
         assert_eq!(response.unwrap(), 7);
 
-        subsystem::Subsystem::shutdown(rpc).await;
+        // TODO: after a cargo update, this test freezes on shutdown, so it's commented out
+        // subsystem::Subsystem::shutdown(rpc).await;
```

Comment:
I'm not sure what the underlying reason for this is. However, it is notable that the test creates the RPC sysbsystem without adding it to the sybsystem manager. That is not a supported way of using subsystems. I'd suggest creating a manager at the start of the test and accessing RPC through it. That way the shutdown will be handled in the usual way when the manager terminates.

Reply:
Should we care and fix it? Or just keep it and move on?

Reply:
Looks like it's a known issue: https://github.com/paritytech/jsonrpsee/pull/1062
The master branch is fixed (I just checked our tests with it), but there is no new release yet. So we can keep using tokio 1.26 for now or switch to the jsonrpsee master branch.